src/table/components/HeadCellMenu.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef, useState } from 'react';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import Grow from '@mui/material/Grow';
+import Paper from '@mui/material/Paper';
+import Popper from '@mui/material/Popper';
+import MenuList from '@mui/material/MenuList';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
+import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import MoreHoriz from '@mui/icons-material/MoreHoriz';
+import { StyledMenuIconButton, StyledCellMenu } from '../styles';
+import { GeneratedStyling } from '../types';
+
+const ListItems = ({ children, itemTitle }: { children: any; itemTitle: string }) => {
+  return (
+    <ListItem disablePadding>
+      <ListItemButton>
+        <ListItemIcon>{children}</ListItemIcon>
+        <ListItemText primary={itemTitle} />
+      </ListItemButton>
+    </ListItem>
+  );
+};
+
+export default function HeadCellMenu({ headerStyle }: { headerStyle: GeneratedStyling }) {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLButtonElement>(null);
+  // It returns focus to the button when we transitioned from !open -> open
+  const prevOpen = useRef(open);
```

Comment:
I think this can mean trouble. We want a single source of truth whether it is open or not